src/SFA.DAS.AssessorService.Application.Api/Services/OrganisationMergingService.cs

```diff
@@ -0,0 +1,280 @@
+                .Where(x => x.State == Microsoft.EntityFrameworkCore.EntityState.Added
+                    || x.State == Microsoft.EntityFrameworkCore.EntityState.Modified
+                    )
+                .Select(x => x.Entity).ToList();
+            */
+
+            // Now save all the changes.
+            try
+            {
+                await _dbContext.SaveChangesAsync();
+            }
+            catch(Exception ex)
+            {
+                // check rollback has worked
+                throw ex;
+            }
+
+            return mergeOrganisation;
+        }
+
+        private MergeOrganisation CreateMergeOrganisations(string primaryEndpointAssessorOrganisationId, string secondaryEndpointAssessorOrganisationId, Guid createdByUserId)
+        {
+            var mo = new MergeOrganisation()
+            {
+                PrimaryEndPointAssessorOrganisationId = primaryEndpointAssessorOrganisationId,
+                SecondaryEndPointAssessorOrganisationId = secondaryEndpointAssessorOrganisationId,
+                CreatedBy = createdByUserId,
+                CreatedAt = DateTime.UtcNow,
+                ApprovedBy = createdByUserId,
+                ApprovedAt = DateTime.UtcNow,
+                Status = MergeOrganisationStatus.Approved,  // auto-approved for now
+            };
+            _dbContext.MergeOrganisations.Add(mo);
+
+            return mo;
+        }
+
+        private void CreateStandardsSnapshot(MergeOrganisation mo, string endpointAssessorOrganisationId, string replicates)
+        {
+            var organisationStandards = _dbContext.OrganisationStandard.Where(e => e.EndPointAssessorOrganisationId == endpointAssessorOrganisationId);
+            foreach (var os in organisationStandards)
+            {
+                mo.MergeOrganisationStandards.Add(CreateMergeOrganisationStandard(replicates, os));
+
+                var organisationStandardVersions = _dbContext.OrganisationStandardVersion.Where(e => e.OrganisationStandardId == os.Id);
+                foreach (var osv in organisationStandardVersions)
+                {
+                    mo.MergeOrganisationStandardVersions.Add(CreateMergeOrganisationStandardVersion(replicates, osv));
+                }
+
+                var organisationStandardDeliveryAreas = _dbContext.OrganisationStandardDeliveryAreas.Where(e => e.OrganisationStandardId == os.Id);
+                foreach (var osda in organisationStandardDeliveryAreas)
+                {
+                    mo.MergeOrganisationStandardDeliveryAreas.Add(CreateMergeOrganisationStandardDeliveryArea(replicates, osda));
+                }
+            }
+        }
+
+        private MergeOrganisationStandard CreateMergeOrganisationStandard(string replicates, OrganisationStandard sourceOrganisationStandard)
+        {
+            var mos = new MergeOrganisationStandard()
+            {
+                EndPointAssessorOrganisationId = sourceOrganisationStandard.EndPointAssessorOrganisationId,
```

Comment:
Part of me does and doesn't like this. Part of me wants to make use of a Mapper, a constructor to encapsulate into the class itself, or to do some form of implicit cast so that the logic is moved out of the service and into the respective objects. But this at least does show you what it's doing so something to talk about maybe?